### PR TITLE
command to safely delete duplicate users

### PR DESCRIPTION
**Pull request by snopoke**
Expansion of what @orangejenny started here: https://dimagi-dev.atlassian.net/browse/ICDS-1091

https://dimagi-dev.atlassian.net/browse/ICDS-1112

### CODE REVIEW

File: corehq/apps/cleanup/management/commands/delete_duplicate_users.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+        for username, user_ids in duplicate_users.items():
+            users_by_id = {
+                user_id: CommCareUser.get_by_user_id(user_id)
+                for user_id in user_ids
+            }
+            safe_user_ids = set(user_ids) - set(user_ids_with_forms)
+
+            if not safe_user_ids:
+                print(f'All users for {username} have forms. Skipping user IDs {user_ids}')
```
Comment (orangejenny) [inline corehq/apps/cleanup/management/commands/delete_duplicate_users.py]: Should this note any duplicate user ids that do have forms? Aren't those users still a problem?

File: corehq/apps/cleanup/management/commands/delete_duplicate_users.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        f = FormES().domain(domain).user_id(user_id).count()
+        if f:
+            users_with_forms.add(user_id)
+    return users_with_forms
+
+
+def delete_duplicate_users(all_users, duplicates_to_delete):
+    all_usernames = {u.username for u in all_users}
+    duplicates_usernames = {u.username for u in duplicates_to_delete}
+    assert len(all_usernames) == 1
```
Comment (orangejenny) [inline corehq/apps/cleanup/management/commands/delete_duplicate_users.py]: Why is this here? Doesn't `all_users` contain all duplicates in the domain?

File: corehq/apps/cleanup/management/commands/delete_duplicate_users.py (hunk 1 above)
Comment (snopoke) [reply corehq/apps/cleanup/management/commands/delete_duplicate_users.py]: This would happen if all duplicate users have form submissions in which case a more complex solution is required that's beyond the scope of this script.

File: corehq/apps/cleanup/management/commands/delete_duplicate_users.py (hunk 2 above)
Comment (snopoke) [reply corehq/apps/cleanup/management/commands/delete_duplicate_users.py]: no, it's all duplicates of a specific user

File: corehq/apps/cleanup/management/commands/delete_duplicate_users.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/cleanup/management/commands/delete_duplicate_users.py]: I don't mean to fix those users, just to let the person running the script know they exist.

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/cleanup/management/commands/delete_duplicate_users.py (hunk 1 above)
Comment (snopoke) [reply corehq/apps/cleanup/management/commands/delete_duplicate_users.py]: I'm not sure I follow, this does output the user IDs.

Comment (orangejenny) [reply corehq/apps/cleanup/management/commands/delete_duplicate_users.py]: But it only outputs them if all duplicate users have forms, right? If there are some duplicates that have forms and some that don't, I'd expect the script to delete the users without forms (like it does) but also to output something like "There are another 10 duplicate users that have forms and therefore weren't deleted, ids are: ..."

Comment (snopoke) [reply corehq/apps/cleanup/management/commands/delete_duplicate_users.py]: I see - I'll update

**Review by orangejenny — APPROVED** [review]
